src/resolve/resolved-dns-packet.c

```diff
@@ -44,20 +44,28 @@ static void rewind_dns_packet(DnsPacketRewinder *rewinder) {
 #define INIT_REWINDER(rewinder, p) do { rewinder.packet = p; rewinder.saved_rindex = p->rindex; } while (0)
 #define CANCEL_REWINDER(rewinder) do { rewinder.packet = NULL; } while (0)
 
-int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
+int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t min_alloc_dsize) {
         DnsPacket *p;
         size_t a;
 
         assert(ret);
 
-        /* When dns_packet_new() is called with mtu == 0, allocate more than the
+        /* The caller may not check what was really allocated, so do not allow to
+         * alloc a DNS paquet bigger than DNS_PACKET_SIZE_MAX.
```

Comment:
*allocate*, *packet*

Hm, the way you phrase the comment is misleading: "was really allocated" is past tense, but we haven't allocated yet.